lib/cf_app_discovery/target.rb

```diff
@@ -40,6 +41,7 @@ def generate_json(index)
       unless paas_metric_exporter?
         data[:labels][:space] = space
       end
+      data[:labels][:__metrics_path__] = "#{path}/metrics" unless path.empty?
```

Comment:
Ok so /metrics is hardcoded here meaning for this to work for the Digital Marketplace we need to get them to change their URL to remove the underscore, correct?

Reply:
Also does this also mean that the marketplace must use basic auth to protect their endpoint and could not use the re-ip-safelist service because this would require them to have two custom routes and we are only taking the first?

Reply:
> Ok so /metrics is hardcoded here meaning for this to work for the Digital Marketplace we need to get them to change their URL to remove the underscore, correct?

yes. this PR is explicitly not about allowing fully custom metrics paths, only about getting the path from the route bound to the app

Reply:
> Also does this also mean that the marketplace must use basic auth to protect their endpoint and could not use the re-ip-safelist service because this would require them to have two custom routes and we are only taking the first?

🤔 yes i think so.  I'm not sure of the right thing to do here.

Reply:
> Also does this also mean that the marketplace must use basic auth to protect their endpoint and could not use the re-ip-safelist service because this would require them to have two custom routes and we are only taking the first?

Does `re-ip-safelist` only recognise the first route registered too?

Reply:
> Does re-ip-safelist only recognise the first route registered too?

You don't bind re-ip-safelist to an app, you bind it directly to a route.  This means that you need to bind it to every route through which you could possibly access the metrics endpoint.

Reply:
`cf bind-route-service cloudapps.digital re-ip-whitelist-service --hostname app-to-protect --path metrics`

Ah yeah, you can give it a path to specifically only whitelist the /metrics.
https://reliability-engineering.cloudapps.digital/monitoring-alerts.html#ip-whitelist-your-applications-metrics-endpoint